modules/pipeline/initialize.go

```diff
@@ -219,6 +229,23 @@ func registerSnippetClient(dbclient *dbclient.Client) error {
 	return nil
 }
 
+func registerClusterHook(bdl *bundle.Bundle) {
+	ev := apistructs.CreateHookRequest{
+		Name:   "pipeline_watch_cluster_changed",
+		Events: []string{bundle.ClusterEvent},
+		URL:    strutil.Concat("http://", conf.PodIP(), endpoints.ClusterHookApiPath),
+		Active: true,
+		HookLocation: apistructs.HookLocation{
+			Org:         "-1",
+			Project:     "-1",
+			Application: "-1",
+		},
+	}
+	if err := bdl.CreateWebhook(ev); err != nil {
+		logrus.Warnf("failed to register watch cluster changed event, %v", err)
```

Comment:
Return err and handle err at invoker side.

Reply:
And I think pipeline should panic if registerClusterHook failed at bootsstrap.

Reply:
Each time  the pod ip is different.
How about old pod ips already registered? Maybe eventbox will continue send msgs to them but failed of course. And the subscribers(pod ips) will be more and more ?

Reply:
If so, maybe we should continue register with service ip and all instances listen from etcd.